tests/integration/helpers.py

```diff
@@ -145,3 +146,43 @@ async def wait_for_mongos_units_blocked(
                 await check_all_units_blocked_with_status(ops_test, db_app_name, status)
     finally:
         await ops_test.model.set_config({hook_interval_key: old_interval})
+
+
+def get_port_from_node_port(ops_test: OpsTest, node_port_name: str) -> None:
+    node_port_cmd = f"kubectl get svc  -n  {ops_test.model.name} |  grep NodePort | grep {node_port_name}"
+    result = subprocess.run(node_port_cmd, shell=True, capture_output=True, text=True)
+    if result.returncode:
+        logger.info("was not able to find nodeport")
+        assert False, f"Command: {node_port_cmd} to find node port failed."
+
+    assert (
```

Comment:
**praise:** I noticed this years ago, but I really like that you have informative messages on your assertions.

Reply:
awh thanks a lot! that means so much to me! <3